test/public/runs/detail.test.js

```diff
@@ -164,30 +161,45 @@ module.exports = () => {
             element.getAttribute('d'))).to.equal(xIconPath);
 
         await pressElement(page, '#edit-run');
-        await waitForTimeout(100);
         await pressElement(page, '#Run-detectors .dropdown-trigger');
-        await waitForTimeout(100);
+        await page.waitForSelector('.dropdown');
+
         expect(await page.$eval(goodQualityRadioSelector, (element) => element.checked)).to.be.false;
         expect(await page.$eval(badQualityRadioSelector, (element) => element.checked)).to.be.true;
+        await pressElement(page, '#save-run');
     });
 
     it('should successfully update end of run reasons', async () => {
-        await reloadPage(page);
         await pressElement(page, '#edit-run');
 
         await page.waitForSelector('#Run-eorReasons select');
         await page.select('#Run-eorReasons select', 'DETECTORS');
-        await waitForTimeout(20);
 
+        await page.waitForSelector('#Run-eorReasons select:nth-child(2)');
         await page.select('#Run-eorReasons select:nth-child(2)', 'CPV');
         await page.type('#Run-eorReasons input', 'A new EOR reason');
-        await page.click('#add-eor-reason');
+        await pressElement(page, '#add-eor-reason', true);
+        try {
+            await page.waitForFunction(
+                () => document.querySelectorAll('#Run-eorReasons .remove-eor-reason').length === 3,
+                { timeout: 5000, polling: 'mutation' },
+            );
+        } catch (e) {
+            await takeScreenshot(page, e);
+        }
+
         // Remove the first EOR reason
-        await page.click('.remove-eor-reason');
+        await pressElement(page, '.remove-eor-reason');
+        await page.waitForFunction(
+            () => document.querySelectorAll('#Run-eorReasons .remove-eor-reason').length === 2,
+            { timeout: 5000, polling: 'mutation' },
```

Comment:
Could you just give it a try without these two in the end? If it make the test pass keep them but maybe put a comment (`Flaky tests randomly fails, this seems to solve the issue`)

Reply:
Done, ended up keeping the options and adding a comment